lib/ansible/modules/cloud/azure/azure_rm_batchaccount_info.py

```diff
@@ -0,0 +1,221 @@
+            description:
+            - The name of the Batch Account.
+            returned: always
+            type: str
+        location:
+            description:
+            - Specifies the supported Azure location where the resource exists.
+            returned: always
+            type: str
+        account_endpoint:
+            description:
+            - The account endpoint used to interact with the Batch service.
+            returned: always
+            type: str
+            sample: sampleacct.westus.batch.azure.com
+        auto_storage_account:
+            description:
+            - Existing storage account with which to associate the Batch Account.
+            returned: always
+            type: str
+            sample: "/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/Microsoft.Storage/storageAccounts/{name}"
+        key_vault:
+            description:
+            - Existing key vault with which to associate the Batch Account.
+            returned: always
+            type: str
+            sample: "/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/Microsoft.KeyVault/vaults/{name}"
+        pool_allocation_mode:
+            description:
+            - The pool acclocation mode of the Batch Account.
+            returned: always
+            type: str
+        tags:
+            description:
+            - A mapping of tags to assign to the batch account.
+            returned: always
+            type: list
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.batch import BatchManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMBatchAccountInfo(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                required=True,
+                type='str'
+            ),
+            name=dict(
+                type='str'
+            ),
+            tags=dict(
+                type='list'
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False
+        )
+        self.mgmt_client = None
+        self.resource_group = None
+        self.name = None
+        self.tags = None
+        super(AzureRMBatchAccountInfo, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(BatchManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        if (self.name):
+            self.results['items'] = self.get()
+        else:
```

Comment:
also need to support list batch account in this subscription, when resource_group is None

Reply:
there's an issue for that: https://github.com/Azure/magic-module-specs/issues/7